snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/SnomedEditingContext.java

```diff
@@ -976,9 +783,24 @@ private Relationship buildDefaultIsARelationship(Concept parentConcept, Concept
 	
 	@Override
 	public void preCommit() {
-		if (deletionPlan != null) {
-			delete(deletionPlan);
+		
+		if (!deletionPlan.isRejected() && !deletionPlan.isEmpty()) {
```

Comment:
Shouldn't we throw an exception if execution reaches this point and the deletion plan is rejected? This will silently skip the deletion parts of the commit.